src/main/java/com/commercetools/pspadapter/paymentHandler/impl/PaymentHandler.java

```diff
@@ -266,27 +285,50 @@ private AddInterfaceInteraction createAddInterfaceInteractionAction(@Nonnull Pay
         return ctpFacade.getPaymentService()
                 // 1. add execute request to the ctp payment as interaction interface
                 .updatePayment(ctpPayment.getId(), Collections.singletonList(interactionAction))
-                .thenCompose(payment -> {
+                .thenCompose(updatedCtpPayment -> {
                     // 2. send execute request to paypal
                     return paypalPlusFacade.getPaymentService().execute(new Payment().setId(paypalPlusPaymentId),
                             paymentExecution)
                             .thenCompose(paypalPayment -> {
-                                AddInterfaceInteraction addInteractionAction = createAddInterfaceInteractionAction(paypalPayment, RESPONSE);
                                 // 3. add paypal response to the ctp payment as interaction interface
-                                return ctpFacade.getPaymentService().updatePayment(payment, Collections.singletonList(addInteractionAction))
+                                return updateCtpPayment(paypalPayment, updatedCtpPayment)
                                         // 4. create charge transaction in the ctp payment
-                                        .thenApply(updatedCtpPayment -> createChargeTransaction(paypalPlusPaymentId, paypalPayment, updatedCtpPayment));
+                                        .thenApply(updatedCtpPayment2 -> createChargeTransaction(paypalPlusPaymentId, paypalPayment, updatedCtpPayment2));
                             });
                 })
                 .thenApply(ignore -> PaymentHandleResponse.ofHttpStatus(HttpStatus.CREATED));
     }
 
+    private CompletionStage<io.sphere.sdk.payments.Payment> updateCtpPayment(Payment paypalPayment, io.sphere.sdk.payments.Payment ctpPayment) {
+        ImmutableList.Builder<UpdateAction<io.sphere.sdk.payments.Payment>> builder = ImmutableList.builder();
+        builder.add(createAddInterfaceInteractionAction(paypalPayment, RESPONSE));
+        PaymentInstruction paymentInstruction = paypalPayment.getPaymentInstruction();
```

Comment:
Why we are sticking with deprecated objects?

> PaymentInstruction object is not available for public use.

Reply:
https://github.com/paypal/PayPal-Java-SDK/issues/313